test/supervisor_test.exs

```diff
@@ -0,0 +1,43 @@
+defmodule KafkaEx.Supervisor.Test do
+  use ExUnit.Case, async: false
+
+  test "uses configured max_restarts" do
+    # stop the application, configure supervision max_restarts and restart the application
+    configured_max_restarts = 15
+
+    :ok = Application.stop(:kafka_ex)
+    Application.put_env(:kafka_ex, :max_restarts, configured_max_restarts)
+    {:ok, _} = Application.ensure_all_started(:kafka_ex)
+    [max_restarts, _max_seconds] = :sys.get_state(KafkaEx.Supervisor) |> elem(9)
```

Comment:
I thought alot about this test, I am using an undocumented API in the test, but I convinced myself it's necessary to guide future refactoring (i.e. some spec that will break if something is wrongly modified)